### PR TITLE
Feat/route create box

### PR DESCRIPTION
**Pull request by lulup52**

### CODE REVIEW

File: server/conf.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const  mysql = require('mysql');
+const  connection = mysql.createConnection({
+host :  'localhost', // adresse du serveur
+user :  'root', // le nom d'utilisateur
+password :  'FEVZNfevzn1.', // le mot de passe
+database :  'tonton_sommelier', // le nom de la base de données
+});
+module.exports = connection;
```
Comment (wasthishelpful) [inline server/conf.js:8]: don't push this... and change your password ;)

File: server/server.js [hunk 2]
```diff
@@ -13,6 +18,16 @@ app.get('/',(req, res) =>{
...
+app.post('/boxes', (req, res) => {
+
+  const formData = req.body;
+  connection.query('INSERT INTO box SET ?', formData, (err, results) => {
+
+    if (err) {
+      console.log(err);
+      res.status(500).send("Erreur lors de la sauvegarde d'une box");
+    } else {
+      res.sendStatus(200).send(res);
```
Comment (wasthishelpful) [inline server/server.js:30]: maybe you can find a better status for response ;)

https://en.m.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success

File: server/server.js [hunk 3]
```diff
@@ -13,6 +18,16 @@ app.get('/',(req, res) =>{
...
-app.post('/',(req, res) =>{
-  res.status(200).send('hello world et tonton');
-})
\ No newline at end of file
+app.post('/boxes', (req, res) => {
+
+  const formData = req.body;
+  connection.query('INSERT INTO box SET ?', formData, (err, results) => {
+
+    if (err) {
```
Comment (wasthishelpful) [inline server/server.js:26]: fix indentation

File: server/server.js [hunk 4]
```diff
@@ -13,6 +18,16 @@ app.get('/',(req, res) =>{
   res.status(200).send('hello world et tonton');
 })
 
-app.post('/',(req, res) =>{
-  res.status(200).send('hello world et tonton');
-})
\ No newline at end of file
+app.post('/boxes', (req, res) => {
+
+  const formData = req.body;
```
Comment (wasthishelpful) [inline server/server.js:23]: you should check if formData effectively contains the data you need. and take a look at 4xx status codes ;)

https://en.m.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors

File: server/tonton_sommelierBDD.sql [hunk 5]
```diff
@@ -1,9 +0,0 @@
-CREATE DATABASE tonton_sommelier;
-
-USE tonton_sommelier;
-
-CREATE TABLE `box`(
-	`id` INT NOT NULL AUTO_INCREMENT,
-	`name` VARCHAR (100) NOT NULL,
-	PRIMARY KEY (`id`)
-);
```
Comment (wasthishelpful) [inline server/tonton_sommelierBDD.sql:9]: why removing the table?

File: server/conf.js (hunk 1 above)
Comment (lulup52) [reply server/conf.js:8]: we have had this file to gitignore, why it was send ?

File: server/tonton_sommelierBDD.sql (hunk 5 above)
Comment (lulup52) [reply server/tonton_sommelierBDD.sql:9]: we keep "archive.js" with all commands but this one is remove when we don't have to update the database

File: conf.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const  mysql = require('mysql');
+const  connection = mysql.createConnection({
+host :  'localhost', // adresse du serveur
+user :  'root', // le nom d'utilisateur
+password :  'FEVZNfevzn1.', // le mot de passe
+database :  'tonton_sommelier', // le nom de la base de données
+});
+module.exports = connection;
```
Comment (wasthishelpful) [inline conf.js:8]: still to remove ;)

take a look at git rm --cached

**Pull request merged**